### DISCUSSION TITLE
v9.0.0-beta7

### DISCUSSION THREAD

**Discussion by mivano** [serilog/serilog-sinks-elasticsearch#443] [Announcements]
## What's Changed
* Create codeql-analysis.yml by @mivano in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/370
* fix: Correct comment about default TypeName by @mikkelbu in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/393
* Proper handling of TypeName = null from appsettings.json by @mariwing in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/420
* Add `ElasticsearchSinkOptions.BufferFileRollingInterval` option by @andrey-kozlov-skuvault in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/416
* Fix internal or IntelliSense typos by @boflynn in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/406
* Bump actions/setup-dotnet from 1 to 2 by @dependabot in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/431
* Bump actions/checkout from 2 to 3 by @dependabot in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/427
* Bump actions/cache from 2 to 3 by @dependabot in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/429
* Bump actions/download-artifact from 2 to 3 by @dependabot in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/424

## New Contributors
* @mikkelbu made their first contribution in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/393
* @mariwing made their first contribution in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/420
* @andrey-kozlov-skuvault made their first contribution in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/416
* @boflynn made their first contribution in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/406
* @dependabot made their first contribution in https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/431

**Full Changelog**: https://github.com/serilog-contrib/serilog-sinks-elasticsearch/compare/v8.4.1+4.build.444...v9.0.0-beta1

<hr /><em>This discussion was created from the release <a href='https://github.com/serilog-contrib/serilog-sinks-elasticsearch/releases/tag/v9.0.0-beta7'>v9.0.0-beta7</a>.</em>

**Comment by bandersen22000** [comment]
Will the final version support .NET framework ? The lib folder does not contain a net45 folder ?

**Reply by bandersen22000** [reply]
Okay just found the Breaking changes :(
**Breaking changes**
Version 9
Dropped support for 456 and sticking now with NETSTANDARD

**Reply by mivano** [reply]
Dropping support is my way to handle the load on this sink. I have not enough time to get support for this code path as well (let alone the build which was broken for months). Happy to accept a PR with support for more standards as long as it rolls out.

**Reply by bandersen22000** [reply]
@mivano Ok. And if I made a pull request for the 8.4.1 version with the changes the changes from [Proper handling of TypeName = null from appsettings.json](https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/420/commits/77e031b49e2e49fae922423fa81c242c60df2780) would that be ok for a version 8.4.2 ?

**Reply by mivano** [reply]
The v9 contains some other merged changes as well. So rather reengineer the needed support into the v9 than having multiple versions to support. I bumped it to v9 as it is a breaking change, the other changes are pretty minor. 
It does need to work with the GitHub actions though so we can roll out packages. 
If you know how and willing to do so, that would be really appreciated.

**Reply by RickSandstrom** [reply]
When is v9 going to be released as a non-beta? Any information about that?

**Comment by DmitriyBobrovskiy** [comment]
hey, can you explain me please why with same configuration, version 8.4.1 doesn't send logs to cloud elastic, but version 9.0.0-beta7 does?
I don't see any description in changelog about this?
thank you

**Reply by RickSandstrom** [reply]
We are also having trouble with 8.4.1 not logging to elastic 8.2, but the v9 beta works.

**Comment by zachawilson** [comment]
Installing v9.0.0-beta7 gives me the following runtime error:
> Could not load file or assembly 'Serilog.Sinks.Elasticsearch' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key.

Is the beta package hosted on nuget.org signed? Are there other alternatives to getting the beta installed in my project?

Thank you.

**Reply by zachawilson** [reply]
Some more info if it helps:

> [HttpException (0x80004005): Could not load file or assembly 'Serilog.Sinks.Elasticsearch, Version=6.0.0.0, Culture=neutral, PublicKeyToken=24c2f752a8e58a10' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)]

```powershell
(dir .\Serilog.Sinks.Elasticsearch.dll).VersionInfo | fl

OriginalFilename  : Serilog.Sinks.Elasticsearch.dll
FileDescription   : Serilog.Sinks.Elasticsearch
ProductName       : Serilog.Sinks.Elasticsearch
Comments          : Serilog sink for Elasticsearch
CompanyName       : Michiel van Oudheusden, Martijn Laarman, Mogens Heller Grabe, Serilog Contributors
FileName          : [fullpath]\Serilog.Sinks.Elasticsearch.dll
FileVersion       : 6.0.0.0
ProductVersion    : 6.0.0-unstable.32+Branch.dev.Sha.7e77498e8afaaec96b508de3e8ec6bd3891b3556
IsDebug           : False
IsPatched         : False
IsPreRelease      : False
IsPrivateBuild    : False
IsSpecialBuild    : False
Language          : Language Neutral
LegalCopyright    : Copyright © Serilog Contributors 2018
LegalTrademarks   :
PrivateBuild      :
SpecialBuild      :
FileVersionRaw    : 6.0.0.0
ProductVersionRaw : 0.0.0.0
```

**Reply by DmitriyBobrovskiy** [reply]
hey, why did you decide that it's v9.0.0-beta7? based on info it's version 6.0.0.
can you provide some sample repository? because in my case everything works as expected without any issues

**Reply by zachawilson** [reply]
@DmitriyBobrovskiy I created an example project. and confirmed that I get the same result in that project, including the listed version 6.0.0

https://github.com/zachawilson/issue-example-serilog-elasticsearch

I'm guessing it is related to me still being on .net 4.7.2 and much of the the world has been able to move on. Please let me know if you have any advice. 


PS - I appreciate the quick response, I'm sorry I didn't see it right away.

**Reply by zachawilson** [reply]
Just for some context. When I use the Visual Studio GUI to switch between 8.4.1 and any of the beta options. I get this error with all the betas and not 8.4.1. The folder name is v9.0.0-beta7 but the dll in the netstandard2.0 folder has the version 6.0.0 info I posted above.

**Reply by DmitriyBobrovskiy** [reply]
It's interesting that I've got the same error but with a bit different package (I was using Rider):
> Could not load file or assembly 'Serilog.Formatting.Elasticsearch' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)

I agree that the problem is probably somehow related to .Net Framework and cannot give you any more details. I think we need info from @mivano.
One thing that I think you can do here is to change security level some way so not strongly signed signatures will be valid. I don't know how to do this though :)

**Reply by zachawilson** [reply]
Thank you for taking a look and confirming. I do get either of those package errors depending on the NuGet configuration.

For now I plan to proceed with 8.4.1 and move my configuration to be setup in code. The primary reason for wanting to pull in the beta is to be able to take advantage of #420 (and any other welcome fixes)

Any input is still appreciated and I'll give suggestions a try and respond.

**Reply by mivano** [reply]
Sorry, it took a while, but we found the culprit. The beta11 packages should be better now, maybe you can validate? Thanks for reporting.

**Reply by zachawilson** [reply]
@mivano Yes I can confirm that this is now acting as expected and the fixes I was looking to take advantage of are working. Thank you 🙏 .

**Reply by mivano** [reply]
Perfect, the 9.0.0-rc1 is the latest and I hope we can promote it to a stable release soon. Thanks for the feedback.
